geth/node/manager.go

```diff
@@ -207,8 +212,13 @@ func (m *NodeManager) Node() (*node.Node, error) {
 	m.RLock()
 	defer m.RUnlock()
 
+	// check if we have a node running
+	if m.node == nil {
+		return nil, ErrNodeOffline
+	}
+
 	// make sure that node is fully started
-	if m.node == nil || m.nodeStarted == nil {
+	if m.nodeStarted == nil {
```

Comment:
Above you use this condition `m.nodeStarted == nil || m.nodeStopped == nil` to return `ErrNoRunningNode`. Is there a reason why it's different here?

Also, if this pattern:
```go
m.RLock()
defer m.RUnlock()

if m.node == nil {
  return nil, ErrNodeOffline
}

if m.nodeStarted == nil || m.nodeStopped == nil {
  return nil, ErrNoRunningNode
}
```
repeats so frequently, maybe it would be possible to extract it to a new private method?